File: app/src/main/java/com/mmjang/ankihelper/ui/ai/AIDictionaryConfigAdapter.java
package com.mmjang.ankihelper.ui.ai;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.button.MaterialButton;
import com.mmjang.ankihelper.R;
import com.mmjang.ankihelper.data.ai.AIDictionaryConfig;

import java.util.List;

public class AIDictionaryConfigAdapter extends RecyclerView.Adapter<AIDictionaryConfigAdapter.DictionaryConfigViewHolder> {
    private List<AIDictionaryConfig> dictionaryConfigList;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(AIDictionaryConfig config);
        void onDeleteClick(AIDictionaryConfig config);
    }

    public AIDictionaryConfigAdapter(List<AIDictionaryConfig> dictionaryConfigList, OnItemClickListener listener) {
        this.dictionaryConfigList = dictionaryConfigList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public DictionaryConfigViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_ai_dictionary_config, parent, false);
        return new DictionaryConfigViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DictionaryConfigViewHolder holder, int position) {
        AIDictionaryConfig config = dictionaryConfigList.get(position);
        holder.bind(config);
    }

    @Override
    public int getItemCount() {
        return dictionaryConfigList.size();
    }

    public void updateList(List<AIDictionaryConfig> newList) {
        this.dictionaryConfigList = newList;
        notifyDataSetChanged();
    }

    class DictionaryConfigViewHolder extends RecyclerView.ViewHolder {
        private TextView textViewDictionaryName;
        private MaterialButton buttonEdit;
        private MaterialButton buttonDelete;

        public DictionaryConfigViewHolder(@NonNull View itemView) {
            super(itemView);
            textViewDictionaryName = itemView.findViewById(R.id.text_view_dictionary_name);
            buttonEdit = itemView.findViewById(R.id.button_edit);
            buttonDelete = itemView.findViewById(R.id.button_delete);
        }

        public void bind(AIDictionaryConfig config) {
            textViewDictionaryName.setText(config.getDictionaryName());

            // Set click listener for the entire item (for editing)
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onItemClick(config);
                    }
                }
            });

            // Set click listener for edit button
            buttonEdit.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onItemClick(config);
                    }
                }
            });

            // Set click listener for delete button
            buttonDelete.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (listener != null) {
                        listener.onDeleteClick(config);
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/lmyby/ankihelper/ui/ai/AIDictionaryConfigAdapter.kt
package com.lmyby.ankihelper.ui.ai

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.button.MaterialButton
import com.lmyby.ankihelper.R
import com.lmyby.ankihelper.data.ai.AIDictionaryConfig

/**
 * Adapter for AI dictionary configuration list
 * Converted to Kotlin as part of Phase 4 adapter migration
 */
class AIDictionaryConfigAdapter(
    private var dictionaryConfigList: List<AIDictionaryConfig>,
    private val listener: OnItemClickListener
) : RecyclerView.Adapter<AIDictionaryConfigAdapter.DictionaryConfigViewHolder>() {

    interface OnItemClickListener {
        fun onItemClick(config: AIDictionaryConfig)
        fun onDeleteClick(config: AIDictionaryConfig)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DictionaryConfigViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_ai_dictionary_config, parent, false)
        return DictionaryConfigViewHolder(view)
    }

    override fun onBindViewHolder(holder: DictionaryConfigViewHolder, position: Int) {
        val config = dictionaryConfigList[position]
        holder.bind(config)
    }

    override fun getItemCount(): Int = dictionaryConfigList.size

    fun updateList(newList: List<AIDictionaryConfig>) {
        dictionaryConfigList = newList
        notifyDataSetChanged()
    }

    inner class DictionaryConfigViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val textViewDictionaryName: TextView = itemView.findViewById(R.id.text_view_dictionary_name)
        private val buttonEdit: MaterialButton = itemView.findViewById(R.id.button_edit)
        private val buttonDelete: MaterialButton = itemView.findViewById(R.id.button_delete)

        fun bind(config: AIDictionaryConfig) {
            textViewDictionaryName.text = config.dictionaryName

            // Set click listener for the entire item (for editing)
            itemView.setOnClickListener {
                listener.onItemClick(config)
            }

            // Set click listener for edit button
            buttonEdit.setOnClickListener {
                listener.onItemClick(config)
            }

            // Set click listener for delete button
            buttonDelete.setOnClickListener {
                listener.onDeleteClick(config)
            }
        }
    }
}
